xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalSession.java

```diff
@@ -96,6 +95,11 @@
      */
     private final Map<String, Object> sessionData = new HashMap<>();
 
+    /**
+     * Software Version (XEP-0092) data as obtained from the peer on this connection.
+     */
+    private Map<String, String> softwareVersionData = new HashMap<>();
+
```

Comment:
I dislike the idea of of maintaining this software version information as a map. XEP-0092 highlights three items; name, version, OS. I'd much rather see a POJO class containing these three items explicitly (perhaps with the IQ or parent Element passed in to the constructor) rather than a Map<String, String> which could hold just about anything.

Reply:
I don't agree. Even though the XEP defines just three elements, there's no predicting what third-party implementations might put in there. This way we display everything advertised.